internal/proxy/task.go

```diff
@@ -1099,6 +1139,84 @@ func (t *alterCollectionTask) OnEnqueue() error {
 	return nil
 }
 
+func hasNamespaceField(schema *schemapb.CollectionSchema) bool {
+	for _, f := range schema.Fields {
+		if f.Name == common.NamespaceFieldName {
+			return true
+		}
+	}
+	for _, f := range schema.StructArrayFields {
+		if f.Name == common.NamespaceFieldName {
+			return true
+		}
+		for _, inner := range f.Fields {
+			if inner.Name == common.NamespaceFieldName {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func hasIsolationProperty(props ...*commonpb.KeyValuePair) bool {
+	for _, p := range props {
+		if p.GetKey() == common.PartitionKeyIsolationKey {
+			return true
+		}
+	}
+	return false
+}
+
+func addNamespaceField(schema *schemapb.CollectionSchema) {
```

Comment:
how about use a fixed system field id(between 2 and 99)?

Reply:
> how about use a fixed system field id(between 2 and 99)?

should we align with `$meta`?